src/pollyCLI/pollyintegration.cpp

```diff
@@ -244,11 +244,18 @@ QVariantMap PollyIntegration::getUserProjectsMap(QByteArray result2) {
 //
 // CALLED FROM: 
 
-
 QVariantMap PollyIntegration::getUserProjects() {
+    QVariantMap userProjects;
     QString getProjectsCommand = "get_Project_names";
     QList<QByteArray> resultAndError = runQtProcess(getProjectsCommand, QStringList() << credFile);
-    QVariantMap userProjects = getUserProjectsMap(resultAndError.at(0));
+    // Putting EpiErrorCode key to detect possible error due to internet connection issues 
+    // or if Polly servers are Down for some reason.
+    if (resultAndError.at(1)!=""){
```

Comment:
Error stream can be populated for n number of reasons. Non empty error stream does not mean that the user is not connected to internet. Error messages like 'package x not found' also get reported on error streams.

That said, Imo the best place to check whether the user is connected to internet or not  would be `index.js`. A quick google search tells me that there is a package called [connectivity](https://www.npmjs.com/package/connectivity) that can be used for this purpose.

Also, If such errors are reported by Node, both GUI and CLI will be able to use them since they use the same Interface i.e `PollyIntegration`